Explorer/Assets/DCL/Navmap/NavmapSearchBarController.cs

```diff
@@ -1,110 +1,153 @@
+        public async UniTask SearchAndShowAsync(CancellationToken ct) =>
+            await navmapBus.SearchForPlaceAsync(currentSearchText, currentPlaceFilter, currentPlaceSorting, ct);
 
-        private void ClickedResult(string coordinates)
+        public void SetText(PlacesData.PlaceInfo place)
         {
-            searchResultPanelController.AnimateLeftRight(true);
-            OnResultClicked?.Invoke(coordinates);
+            view.inputField.SetTextWithoutNotify(place.title);
         }
 
-        private void OnValueChanged(string searchText)
+        public void ClearSearch(bool notify = false)
+        {
+            if (notify)
+                view.inputField.text = "";
+            else
+                view.inputField.SetTextWithoutNotify("");
+
+            view.clearSearchButton.gameObject.SetActive(false);
+        }
+
+        public void EnableBack()
+        {
+            view.BackButton.gameObject.SetActive(true);
+        }
+
+        public void DisableBack()
+        {
+            view.BackButton.gameObject.SetActive(false);
+        }
+
+        private void OnBackClicked()
+        {
+            backCancellationToken = backCancellationToken.SafeRestart();
+            navmapBus.GoBackAsync(backCancellationToken.Token).Forget();
+        }
+
+        private void ClickedHistoryResult(string historyText)
+        {
+            view.inputField.SetTextWithoutNotify(historyText);
+            OnInputValueChanged(historyText);
+            historyRecordPanelView.gameObject.SetActive(false);
+        }
+
+        private void OnInputValueChanged(string searchText)
         {
             view.clearSearchButton.gameObject.SetActive(!string.IsNullOrEmpty(searchText));
-            OnSearchTextChanged?.Invoke();
-            cts.SafeCancelAndDispose();
-            cts = new CancellationTokenSource();
+            searchCancellationToken = searchCancellationToken.SafeRestart();
 
-            if (string.IsNullOrEmpty(searchText) || searchText.Length < 3)
+            if (string.IsNullOrEmpty(searchText))
             {
-                searchResultPanelController.Hide();
                 historyRecordPanelView.gameObject.SetActive(true);
                 return;
             }
+
             historyRecordPanelView.gameObject.SetActive(false);
 
-            // Suppress cancellation but let other exceptions be printed
-            SearchAndShowAsync(searchText).SuppressCancellationThrow().Forget();
-        }
+            currentSearchText = searchText;
 
-        public void ResetSearch()
-        {
-            searchResultPanelController.Reset();
-            ClearSearch();
+            SearchAsync(searchCancellationToken.Token)
+               .Forget();
+
+            return;
+
+            async UniTaskVoid SearchAsync(CancellationToken ct)
+            {
+                await UniTask.Delay(1000, cancellationToken: ct).SuppressCancellationThrow();

```

Comment:
this was a leftover by me sorry, can you make the 1000 a constant with a descriptive name?